src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -1002,7 +1002,98 @@ class segment_meta_cstore::impl
         _write_buffer.erase(it);
     }
 
-    void insert(const segment_meta& m) {
+    bool can_be_inserted(const segment_meta& m) const {
```

Comment:
Feels like there should be some unit tests to accompany this method specifically (at least to cover the cases in this comment)